rasterio/rio/warp.py

```diff
@@ -63,11 +64,47 @@
               help='align the output bounds based on the resolution')
 @options.overwrite_opt
 @options.creation_options
+@click.option(
+    "--wo",
+    "--warper-option",
```

Comment:
This option is untested. Any concerns about that?

Reply:
Indeed, reproject doesn't distinguish between warper and transformer options. This may (will?) bite us some day if GDAL ever overloads the option names because we use the same option set for each. Let's not doing anything about that now.

For the CLI, I think we can make onboarding smoother if we have `--wo` and `--to` options like gdalwarp, but then combine them at the option parsing level for now.